src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -1,122 +1,264 @@
 package frc.robot.subsystems;
 
 import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkBase.IdleMode;
 import com.revrobotics.CANSparkLowLevel.MotorType;
-import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxSim;
+import com.revrobotics.RelativeEncoder;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
 import frc.robot.Constants;
+import frc.robot.Robot;
 import frc.robot.utilities.DebugEntry;
 
 public class ShooterSubsystem extends SubsystemBase {
 
-  private final CANSparkMax shooterTopMotor;
-  private final CANSparkMax shooterBottomMotor;
+  private final CANSparkMaxSim shooterLeftMotor;
+  private final CANSparkMaxSim shooterRightMotor;
 
-  private DebugEntry<Double> topMotorSpeedEntry;
-  private DebugEntry<Double> bottomMotorSpeedEntry;
-  private DebugEntry<Double> topMotorTargetSpeedEntry;
-  private DebugEntry<Double> bottomMotorTargetSpeedEntry;
+  private final RelativeEncoder shooterLeftMotorEncoder;
+  private final RelativeEncoder shooterRightMotorEncoder;
+
+  private ShuffleboardTab ShooterTab = Shuffleboard.getTab("Shooter Tab");
+
+  private DebugEntry<Double> leftMotorOutputEntry;
+  private DebugEntry<Double> leftMotorSpeedEntry;
+  private DebugEntry<Double> leftMotorTargetSpeedEntry;
+  private DebugEntry<Double> leftMotorTemperatureEntry;
+
+  private DebugEntry<Double> rightMotorOutputEntry;
+  private DebugEntry<Double> rightMotorSpeedEntry;
+  private DebugEntry<Double> rightMotorTargetSpeedEntry;
+  private DebugEntry<Double> rightMotorTemperatureEntry;
+
+  private DebugEntry<Double> leftFlywheelInputEntry;
+  private DebugEntry<Double> leftFlywheelAngularVelocityEntry;
+
+  private DebugEntry<Double> rightFlywheelInputEntry;
+  private DebugEntry<Double> rightFlywheelAngularVelocityEntry;
+
+  private DebugEntry<Boolean> shooterReadyEntry;
+
+  private ShuffleboardTab shooterTab = Shuffleboard.getTab("Shooter fire");
```

Comment:
Remove this tab and use the other one on line 31

Reply:
Resolved in 677ed4c